posthog/hogql/database/schema/persons.py

```diff
@@ -46,72 +44,79 @@ def select_from_persons_table(
+    if (
+        node.where
+        and context.team
+        and posthoganalytics.feature_enabled(
+            "persons-inner-where-optimization",
+            str(context.team.uuid),
+            groups={"organization": context.team.organization_id},
+            group_properties={"organization": {"id": context.team.organization_id}},
+            only_evaluate_locally=True,
+            send_feature_flag_events=False,
+        )
+    ):
+        inner_select = cast(
+            ast.SelectQuery,
+            parse_select(
+                """
+            SELECT id
+            FROM raw_persons as inner_where_optimization
             """
             ),
         )
-        select.settings = HogQLQuerySettings(optimize_aggregation_in_order=True)
-        if filter is not None:
-            cast(ast.SelectQuery, cast(ast.CompareOperation, select.where).right).where = filter
-
-        for field_name, field_chain in join_or_table.fields_accessed.items():
-            # We need to always select the 'id' field for the join constraint. The field name here is likely to
-            # be "persons__id" if anything, but just in case, let's avoid duplicates.
-            if field_name != "id":
-                select.select.append(
-                    ast.Alias(
-                        alias=field_name,
-                        expr=ast.Field(chain=field_chain),
-                    )
-                )
-    else:
-        select = argmax_select(
-            table_name="raw_persons",
-            select_fields=join_or_table.fields_accessed,
-            group_fields=["id"],
-            argmax_field="version",
-            deleted_field="is_deleted",
-            timestamp_field_to_clamp="created_at",
-        )
-        select.settings = HogQLQuerySettings(optimize_aggregation_in_order=True)
-        if filter is not None:
-            if select.where:
-                select.where = And(exprs=[select.where, filter])
-            else:
-                select.where = filter
-
-    if context.modifiers.optimizeJoinedFilters:
-        extractor = WhereClauseExtractor(context)
+        extractor = WhereClauseExtractor(context, is_join=is_join)
         extractor.add_local_tables(join_or_table)
         where = extractor.get_inner_where(node)
+
         if where and select.where:
-            select.where = And(exprs=[select.where, where])
-        elif where:
-            select.where = where
+            inner_select.where = where
+            cast(ast.SelectQuery, cast(ast.CompareOperation, select.where).right).where = ast.CompareOperation(
+                left=ast.Field(chain=["id"]), right=inner_select, op=ast.CompareOperationOp.In
+            )
+
+    if filter is not None:
+        if select.where:
+            cast(ast.SelectQuery, cast(ast.CompareOperation, select.where).right).where = ast.And(
+                exprs=[select.where, filter]
+            )
```

Comment:
I think this is where the reference loop is being created that is failing the tests — `select.where` ends up pointing to itself as `select.where.right.where.exprs[0]`.